packages/babel-plugin-transform-react-jsx-self/src/index.js

```diff
@@ -16,11 +16,78 @@ import { types as t } from "@babel/core";
 
 const TRACE_ID = "__self";
 
+/**
+ * Returns whether the given path contains a `super()` call.
+ */
+function containsSuperCall(sourcePath) {
+  // We don't want to go into inner classes as their super() calls aren't relevant for us.
+  if (sourcePath.isClassDeclaration() || sourcePath.isClassExpression()) {
+    return false;
+  }
+  let exists = false;
+  sourcePath.traverse({
+    // Again, we don't go into inner classes.
+    ClassDeclaration(path) {
+      path.skip();
+    },
+    ClassExpression(path) {
```

Comment:
Can hook to the `Class` alias which includes both `ClassDeclaration` and `ClassExpression`.

Reply:
Done.